docs/xxadaptivegcthreading.md

```diff
@@ -0,0 +1,60 @@
+<!--
+* Copyright (c) 2017, 2021 IBM Corp. and others
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution and is available at
+* https://www.eclipse.org/legal/epl-2.0/ or the Apache
+* License, Version 2.0 which accompanies this distribution and
+* is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* This Source Code may also be made available under the
+* following Secondary Licenses when the conditions for such
+* availability set forth in the Eclipse Public License, v. 2.0
+* are satisfied: GNU General Public License, version 2 with
+* the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] http://openjdk.java.net/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# -XX:\[+|-\]AdaptiveGCThreading
+
+When this option is enabled, the active GC thread count is adjusted for each collection cycle based on heuristics. That is, once a collection cycle successfully completes, the collector evaluates parallelism using aggregated thread busy/stall statistics gathered during the completed cycle and projects a new thread count for the next cycle. For example, the thread count may be reduced if it is determined that unnecessary overhead was incurred as a result of synchronization, lack of work sharing or CPU availability. Similarly, the thread count may be increased if there's an opportunity to gain benefits from increased parallelism. 
```

Comment:
Can we say "garbage collection (GC) cycle" .....  then easier to say "when a GC cycle successfully completes" etc.

I think the words "busy/stall" are not necessary, simply saying "aggregated thread statistics" is sufficient, I think.

Please correct/add commas for clarity:

For example, the thread count might be reduced if it is determined that *an* unnecessary overhead was incurred as a result of synchronization, lack of work sharing, or CPU availability.